src/auth/User.cc

```diff
@@ -47,6 +47,27 @@ Auth::User::credentials(CredentialState newCreds)
     credentials_state = newCreds;
 }
 
+void
+Auth::User::noteHelperTtl(const char *ttlNote)
+{
+    if (ttlNote) {
+        const Auth::Ttl noteValue(strtoll(ttlNote, nullptr, 10));
+        updateExpiration(noteValue);
+    } else {
+        updateExpiration(config->credentialsTtl);
+    }
+}
+
+void
+Auth::User::updateExpiration(const Auth::Ttl &ttl)
+{
+    // prevent signed overflows in time calculation
+    if (ttl.count() >= (std::numeric_limits<decltype(current_time.tv_sec)>::max() - current_time.tv_sec))
+        expiretime = std::numeric_limits<decltype(current_time.tv_sec)>::max();
+    else
+        expiretime = current_time.tv_sec + ttl.count();
```

Comment:
This code probably does prevent overflows for the sum, as claimed, but does not prevent overflows in the assignment to `expiretime` when `time_t` (i.e. `tv_sec`) can hold values larger than `long` (i.e. `expiretime`) can. I need more time if you want me to suggest a good fix for this problem.

<!-- https://godbolt.org/z/f3PorfM8z -->